### ISSUE TITLE
Loading encrypted/read-only  libraries specified by "uses" fails.

### ISSUE THREAD

**Issue by dietmarw** [OpenModelica/OpenModelica#9760]
### Description

When a library points to an encrypted or simply read-only library in `mol` 

### Steps to Reproduce

1. Unzip [MWE_mol.zip](https://github.com/OpenModelica/OpenModelica/files/10092740/MWE_mol.zip) to a test directory. 
2. open `A/package.mo` 
3. you will notice that only `B` will be opened but `C` is not recognised. 

Adding the test directory to `MODELICAPATH` does not help.

### Expected Behavior

A, B, C should be loaded. 

It could well be that this behaviour is related to the inability of OMEdit to load other packages that are in the same source directory when not saved in the directory format but file format instead. Which I think is also a bug but not 100% sure if related or not.

**Comment by casella** [comment]
@dietmarw correct me if I'm wrong, but I'm afraid the problem here is that the whole package encryption issue is not yet standardized. There is work going on, see e.g. modelica/ModelicaSpecification#1868, but it's not done yet. Files with .mol extension are not even mentioned in the MLS, not event in the current [3.6 draft](https://specification.modelica.org/master/MLS.pdf).

I'm not sure what should actually be the intended behaviour.

Maybe @niklwors can comment on that.

**Comment by dietmarw** [comment]
@casella This is not a question of the encryption standardization but of the support of OpenModelica's own formats. OpenModelica offers the nice feature of saving packages in `mol` format (also without encryption as read-only package). The problem is that it does not support the loading of its own format the way it supports the loading of normal Modelica packages. That's what this ticket is about.

**Comment by casella** [comment]
Fair enough. We should then update the class loader to also look for .mol files in the MODELICAPATH. @perost could you take care of that?

**Comment by perost** [comment]
> Fair enough. We should then update the class loader to also look for .mol files in the MODELICAPATH. @perost could you take care of that?

There's an old PR, #8313, that implements loading of .mol-files. But it's not merged since there's some open questions about how to actually do it properly, and some parts needs to be improved.

**Comment by dietmarw** [comment]
At the same time, the loading of packages in single file format can be fixed. Currently, only `A/package.mo` gets loaded automatically but not `A.mo`.

**Comment by casella** [comment]
> At the same time, the loading of packages in single file format can be fixed. Currently, only `A/package.mo` gets loaded automatically but not `A.mo`.

Ops. @perost can you please check that.

**Comment by casella** [comment]
@dietmarw do you still get the same problem?

**Comment by dietmarw** [comment]
@casella It is actually more broken now since opening `A/package.mo` will not even open anything else. Should also be easy to test for your devs follwing the instructions given in the "Steps to reproduce".

**Comment by casella** [comment]
@adeas31 please check at your earliest convenience.
